### DISCUSSION TITLE
给QNAP开发插件

### DISCUSSION THREAD

**Discussion by iranee** [messense/aliyundrive-webdav#166] [Show and tell]
这几天给QNAP打包了个同样的Aliyunpan webdav插件，用的是Go语言写的那个开源。发现威联通本身加载不了，但是第三方可以加载，作者也正在改BUG。

我想问下，这个相较于Go语言版，有什么大的区别和优劣势？我想了解下，看看哪个更适合威联通。

我看到你有写开发初衷，很赞！

**Comment by messense** [comment]
大同小异，Rust 写的相比 Java/Golang 版本无 gc，理论上资源占用会小一些。

**Comment by iranee** [comment]
已经在威联通上测试通过，这两天改下，再搞个Token过期推送通知。

**Reply by messense** [reply]
FYI，应该可以直接通过 HTTP 访问 `http://$user:$password@$host:$port/` 检查状态码是否为 2xx 确定当前在使用 token 是否有效。

**Reply by messense** [reply]
哦对了，上面说的这个检测方法需要启用 `--auto-index` 命令行参数。

**Reply by iranee** [reply]
先看下哪种容易检测出结果。
我发现如果Token不正确的话，会有ERROR错误提示

**Reply by messense** [reply]
你觉得 OK 的话，我可以把你的项目链接加到 [readme](https://github.com/messense/aliyundrive-webdav#qnap-%E5%A8%81%E8%81%94%E9%80%9A-nas) 上。

**Reply by iranee** [reply]
可以可以![3DA77C72](https://user-images.githubusercontent.com/19551701/143410321-4cb267ed-cfab-408c-8c7d-a9ec1357e01f.png)

**Reply by messense** [reply]
Done. https://github.com/messense/aliyundrive-webdav#qnap-%E6%8F%92%E4%BB%B6

**Reply by iranee** [reply]
外贸人新手撸码，慢慢来。

**Reply by iranee** [reply]
https://github.com/iranee/qnap-aliyunpan-webdav/issues/8

你好，我这里经常遇到有人询问，aliyundriver可执行文件被系统扫描并删除

威联通NAS，内置开源杀毒软件CLamAV

**Reply by messense** [reply]
误报没啥办法，只能提交给 QNAP 官方。
